clang/test/SemaCXX/warn-thread-safety-analysis.cpp

```diff
@@ -2249,7 +2249,7 @@ void fooF1(Foo *f) EXCLUSIVE_LOCKS_REQUIRED(f->mu_) {
   f->a = 1;
 }
 
-void fooF2(Foo *f);
+void fooF2(Foo *f); // expected-warning {{attribute mismatch between function declarations of 'fooF2'}}
 void fooF2(Foo *f) EXCLUSIVE_LOCKS_REQUIRED(f->mu_) {
```

Comment:
Shouldn't the warning be on the second declaration with a note on the first declaration?

Reply:
I guess that would make a little more sense, but swapping them also causes the note to not get emitted somehow.

Reply:
@aaronpuchert Is the current order fine or do you want me to investigate the missing note?